Solutions/Corvus.Leasing.Azure/Microsoft/Extensions/DependencyInjection/AzureLeaseProviderServiceCollectionExtensions.cs

```diff
@@ -21,9 +21,10 @@ public static class AzureLeaseProviderServiceCollectionExtensions
         /// Add the Azure implementation of leasing to the service collection.
         /// </summary>
         /// <param name="services">The service collection to which to add azure leasing.</param>
+        /// <param name="options">The configuration options.</param>
         /// <param name="configureLeasing">An optional configuration function for leasing.</param>
         /// <returns>The service collection.</returns>
-        public static IServiceCollection AddAzureLeasing(this IServiceCollection services, Action<AzureLeaseProvider> configureLeasing = null)
+        public static IServiceCollection AddAzureLeasing(this IServiceCollection services, AzureLeaseProviderOptions options, Action<AzureLeaseProvider> configureLeasing = null)
```

Comment:
This has the same problem as with the similar changes to tenancy (which I hadn't realised until reviewing these PRs): the way the Azure Functions host loads config makes life awkward for us. You can't actually get your hands on the config settings until after DI setup has completed.

So we really need to add methods that have signatures along the lines of 

```csharp
public static IServiceCollection AddAzureLeasing(this IServiceCollection services, Func<IServiceProvider, AzureLeaseProviderOptions> getOptions, Action<AzureLeaseProvider> configureLeasing = null)
```

so that DI init can call it like this:

```csharp
services.AddAzureLeasing(sp => sp.GetRequiredService<IConfiguration>().Get<AzureLeaseProviderOptions>();
```

because the Functions SDK doesn't give you any other supported way to do it.

(One of the attractions of the way we use to load settings in a completely unsupported way is that it avoided this problem...)

Reply:
We've added this extension in the latest update.